src/main/kotlin/com/chattriggers/ctjs/minecraft/libs/renderer/Shape.kt

```diff
@@ -100,24 +100,22 @@ class Shape(private var color: Int) {
     }
 
     fun draw() = apply {
-        val a = (color shr 24 and 255).toFloat() / 255.0f
-        val r = (color shr 16 and 255).toFloat() / 255.0f
-        val g = (color shr 8 and 255).toFloat() / 255.0f
-        val b = (color and 255).toFloat() / 255.0f
-
         val tessellator = MCTessellator.getInstance()
         val worldRenderer = tessellator.getRenderer()
 
         GlStateManager.enableBlend()
         GlStateManager.disableTexture2D()
         GlStateManager.tryBlendFuncSeparate(770, 771, 1, 0)
-        if (Renderer.colorized == null)
-            GlStateManager.color(r, g, b, a)
+        Renderer.doColor(color.toLong())
 
         worldRenderer.begin(drawMode, DefaultVertexFormats.POSITION)
 
-        for (vertex in vertexes)
-            worldRenderer.pos(vertex.x.toDouble(), vertex.y.toDouble(), 0.0).endVertex()
+        if (!Renderer.isCounterClockwise(vertexes))
+            vertexes.reverse()
```

Comment:
I think it would be better to store a potentially-reversed view of the list instead of doing this check every frame. There might be a reversed list view class somewhere in the stdlib, but if not it should be easy enough to make one. It would have to be rechecked every time the vertex area is mutated.